src/views/ShareCredentialSelect/ShareCredentialSelect.tsx

```diff
@@ -0,0 +1,176 @@
+import { useState, useEffect } from 'react';
+import { browser } from 'webextension-polyfill-ts';
+import { Link, useLocation } from 'react-router-dom';
+import { sortBy } from 'lodash-es';
+
+import { Identity } from '../../utilities/identities/types';
+import { useIdentities } from '../../utilities/identities/identities';
+import { useIdentityCredentials } from '../../utilities/credentials/credentials';
+import { usePopupData } from '../../utilities/popups/usePopupData';
+import { ShareInput } from '../../channels/shareChannel/types';
+
+import { paths } from '../paths';
+
+import { useAddressBalance } from '../../components/Balance/Balance';
+import { KiltAmount } from '../../components/KiltAmount/KiltAmount';
+import { ShareCredentialCard } from '../../components/CredentialCard/ShareCredentialCard';
+import { Stats } from '../../components/Stats/Stats';
+import { Avatar } from '../../components/Avatar/Avatar';
+import { Selected } from '../ShareCredential/ShareCredential';
+
+import * as styles from './ShareCredentialSelect.module.css';
+
+function MatchingIdentityCredentials({
+  identity,
+  onSelect,
+  selected,
+  match,
+}: {
+  identity: Identity;
+  onSelect: (value: Selected) => void;
+  selected?: Selected;
+  match: () => void;
+}): JSX.Element | null {
+  const data = usePopupData<ShareInput>();
+
+  const { credentialRequest } = data;
+
+  const { cTypes } = credentialRequest;
+  const cTypeHashes = cTypes.map(({ cTypeHash }) => cTypeHash);
+
+  const credentials = useIdentityCredentials(identity.did);
+
+  const matchingCredentials = credentials?.filter(
+    (credential) =>
+      cTypeHashes.includes(credential.request.claim.cTypeHash) &&
+      credential.request.claim.owner === identity.did,
```

Comment:
I think here we must use that to handle the DID upgrades
```suggestion
      parseDidUrl(credential.request.claim.owner).fullDid === parseDidUrl(identity.did).fullDid,
```

Reply:
Oh, good catch!